### DISCUSSION TITLE
adding childImageSharp nodes to images from strapi v4

### DISCUSSION THREAD

**Discussion by unexpand** [gatsbyjs/gatsby#36861] [Help] [answered]
I am trying to add childImageSharp nodes to the image url which is in nested data under cover_image.
I am intersted in understanding how to customize data inside an array using **gatsby-node.js** . I have tried many ways but I am unable to download data into my **src/data** folder , so as to use the images using GatsbyImage.

The issue I am having is linking data from the array , can please point me how to achieve this. 

I have tried using gatsby-node.js and createResolvers and createSchemaCustomization, but nothing works with this nested data.

My query looks like this and the data is coming from Strapi v4 generated graphql schema. 

cmsData -> data -> posts -> [data] -> attributes ->  cover_image -> data -> attributes -> alternativeText - url - name - width - height

 {
      resolve: `gatsby-source-graphql`,
      options: {
        typeName: `CmsData`,
        fieldName: `cmsData`,
        url: 'https://myurl.url/graphql',
        headers: {
          Authorization: `Bearer sometoken`
        }
      },
    },

`
exports.sourceNodes = ({ actions, createNodeId, createContentDigest, getNodesByType }) => {
  const { createNode } = actions
  const myData = getNodesByType("CmsData")
  const nodeContent = JSON.stringify(myData)
  console.log(nodeContent);
  const nodeMeta = {
   urlImage: myData.url
  }

  const node = Object.assign({}, myData, nodeMeta)
  createNode(node)
}

`

Since CmsData type is specifed , shouldn't I be able to access it using getNodesByType("CmsData").

Is this a bug if I can't access my generated graphql ?

`query{
  posts{
    data{
      attributes{
        cover_image{
          data{
            attributes{
              alternativeText
              url
              name
              width
              height
            }
          }
        }
      }
    }
  }
}`

Query result looks like this 

`{
  "data": {
    "posts": {
      "data": [
        {
          "attributes": {
            "cover_image": {
              "data": {
                "attributes": {
                  "alternativeText": "Gatsby-Logo.svg",
                  "url": "/uploads/Gatsby_Logo_7a08acf048.svg",
                  "name": "Gatsby-Logo.svg",
                  "width": 106,
                  "height": 28
                }
              }
            }
          }
        },
        {
          "attributes": {
            "cover_image": {
              "data": {
                "attributes": {
                  "alternativeText": "tailwindcss-logo.svg",
                  "url": "/uploads/tailwindcss_logo_747eafa398.svg",
                  "name": "tailwindcss-logo.svg",
                  "width": 262,
                  "height": 33
                }
              }
            }
          }
        }
      ]
    }
  }
}`

**Comment by LekoArts** [comment, marked as answer]
`gatsby-source-graphql` has known limitations: https://www.gatsbyjs.com/plugins/gatsby-source-graphql/#known-limitations
One of them being that you can't download the images and have image optimizations.

You should use https://www.gatsbyjs.com/plugins/gatsby-source-strapi/
